libs/ui-lib/lib/ocm/components/clusterConfiguration/platformIntegration/ExternalPlatformDropdown.tsx

```diff
@@ -43,8 +44,29 @@ export type ExternalPlatformInfo = {
   disabledReason?: string;
 };
 
+const getDisabledReasonForExternalPlatform = (
+  featureId: FeatureId,
+  isSNO: boolean,
+  isNutanixOrVsphere: boolean,
+  label: string,
+  newFeatureSupportLevelContext: NewFeatureSupportLevelData,
+  featureSupportLevelData?: NewFeatureSupportLevelMap | null,
+  cpuArchitecture?: string,
+): string | undefined => {
```

Comment:
I suggest you only send a `platform: PlatformType` as a paramenter instead of `featureId`, `isNutanixOrVsphere`, and `label`. The `platform` parameter can be used to calculate all three of the value in the function if necessary.